classes/Language.php

```diff
@@ -1262,6 +1263,30 @@ public static function installLanguagePack($iso, $params, &$errors = [])
         return count($errors) ? $errors : true;
     }
 
+    private static function updateCurrenciesCldr(Language $language)
+    {
+        /** @var Currency[] $currencies */
+        $currencies = Currency::getCurrencies(true, false, false);
+        $container = SymfonyContainer::getInstance();
+        /** @var LocaleRepository $localeRepoCLDR */
+        $localeRepoCLDR = $container->get('prestashop.core.localization.cldr.locale_repository');
+
+        foreach ($currencies as $currency) {
+            $names = $currency->getLocalizedNames();
+            $symbols = $currency->getLocalizedSymbols();
+
+            $cldrCurrency = $localeRepoCLDR->getLocale($language->locale)->getCurrency($currency->iso_code);
```

Comment:
You can extract this : `$localeRepoCLDR->getLocale($language->locale)` from the loop foreach.

Reply:
Done